alns/accept/tests/test_worse_accept.py

```diff
@@ -0,0 +1,163 @@
+import pytest
+
+import numpy as np
+import numpy.random as rnd
+from numpy.testing import assert_, assert_equal, assert_raises
+
+from alns.accept import WorseAccept
+from alns.tests.states import Zero, One, Two
+
+
+class MockRNG:
+    """
+    Mock Random Number Generator.
+    """
+
+    def __init__(self, values):
+        self.values = values
+
+    def random(self):
+        return self.values.pop(0)
+
+
+def test_raises_invalid_parameters():
+    """
+    Worse accept does not work with non-probabilistic start and end probability
+    so those should not be accepted. Negative step values should also not be
+    accepted.
+    """
+    with assert_raises(ValueError):  # start prob cannot be negative
+        WorseAccept(-1, 1, 0.1)
+
+    with assert_raises(ValueError):  # start prob cannot be larger than 1
+        WorseAccept(2, 1, 0.1)
+
+    with assert_raises(ValueError):  # end prob cannot be negative
+        WorseAccept(1, -1, 0.1)
+
+    with assert_raises(ValueError):  # end prob cannot be larger than 1
+        WorseAccept(1, -1, 0.1)
+
+    with assert_raises(ValueError):  # updating step cannot be negative
+        WorseAccept(1, 1, -0.1)
```

Comment:
Refactor: this is a perfect candidate for a parametrized method.

Reply:
Here I specifically ask for a parametrised test, because when those fail, they give much more info about what caused the failure - particularly the inputs.

Reply:
I parametrized the test `test_raises_invalid_paramters` and added a new parametrized test `test_no_raise_valid_parameters`. I removed other tests that were checking for invalid/valid parameters and put them as parameters in the new tests.